src/core_plugins/kibana/public/visualize/saved_visualizations/saved_visualizations.js

```diff
@@ -32,9 +29,7 @@ app.service('savedVisualizations', function (Promise, kbnIndex, SavedVis, Privat
     }
 
     if (!typeName || !visTypes.byName[typeName]) {
-      if (!typeName) notify.error('Visualization type is missing. Please add a type to this visualization.', source);
-      else notify.error('Visualization type of "' + typeName + '" is invalid. Please change to a valid type.', source);
-      return kbnUrl.redirect('/management/kibana/objects/savedVisualizations/{{id}}', { id: source.id });
+      return null;
```

Comment:
I'm worried that we are hiding an invalid state from the user, with no error message and hiding the bad visualizations.  There is no way to view their contents, to change the type, or to delete them.

What if we kept the notify.error, maybe changing it something like:
```
      if (!typeName) {
        notify.error(`Visualization ${source.title} is missing type. Please add a type, or delete this visualization.`, source);
      } else {
        notify.error(`Visualization ${source.title} with type of "${typeName}" is invalid. Please change to a valid type or delete the visualization.`, source);
      }
```

Then we return source instead of null (and skip the filtering on null)?  Then you'll get the error message, but you can still delete the invalid visualizations, or ignore the error for the time being (instead of the previous redirect which forced you to handle the error before doing anything else).

Reply:
wouldn't that generate 100s of errors (in case you have 100 invalid visualizations?)
also the notification would keep coming back everytime you went to visualize ( i am guessing here ).

however you are right, they should not just disappear

rethinking the behaviour (let me know if you agree):
- on existing dashboards the panel should show the error, like "invalid visualization type"
- on creating new dashboards, visualization should not show in the list of visualizations
- on visualize page, should the visualization be visible ? probably yes, but somehow indicated that its broken at the moment? or maybe we should completely hide it here as well ?
- opening a visualization should redirect back to visualize home page (as we cant open it)
- visualization should be visible under saved object , where it can be edited/deleted

Reply:
`on visualize page, should the visualization be visible ? probably yes, but somehow indicated that its broken at the moment? or maybe we should completely hide it here as well ?`

fwiw, I have slight preference for hiding.

Def in agreement with your other recommendations.

Reply:
Thinking about it more i agree we should hide it on visualizations page. This is a page that every user hits, it would probably be confusing for some to see visualizations they can't open.

Reply:
I defer to you both.  If people become confused we can always reconsider.  Probably not something that many people will run into anyway.